afpacket/afpacket.go

```diff
@@ -248,16 +250,17 @@ func (h *TPacket) releaseCurrentPacket() error {
 func (h *TPacket) ZeroCopyReadPacketData() (data []byte, ci gopacket.CaptureInfo, err error) {
 	h.mu.Lock()
 retry:
-	if h.current == nil || !h.current.next() {
+	if h.current == nil || !h.headerNextNeeded || !h.current.next() {
 		if h.shouldReleasePacket {
 			h.releaseCurrentPacket()
 		}
 		h.current = h.getTPacketHeader()
```

Comment:
@gconnell it seems to me that getTPacketHeader returns a new position depending on h.offset and h.offset is moved forward by releaseCurrentPacket which is protected by shouldReleasePacket. shouldReleasePacket is set to true only when on poll success ?